### PR TITLE
Add minimal symlink support via GHContent.getTarget()

### PR DESCRIPTION
**Pull request by jtnord**
# Description 

this adds rudimentary symlink support (just enough to get a symlink).

still todo, howto handle GHRepository.getFileContent where the path
contains a symlink within the repository

Related to #875 

# Before submitting a PR:
We love getting PRs, but we hate asking people for the same basic changes every time. 

- [x] Push your changes to a branch other than `master`. Create your PR from that branch.    
- [x] Add JavaDocs and other comments
- [x] Write tests that run and pass in CI. See [CONTRIBUTING.md](CONTRIBUTING.md) for details on how to capture snapshot data.
- [x] Run `mvn clean compile` locally. This may reformat your code, commit those changes.
- [ ] Run `mvn -D enable-ci clean install site` locally. If this command doesn't succeed, your change will not pass CI.

# When creating a PR: 

- [x] Fill in the "Description" above. 
- [x] Enable "Allow edits from maintainers".

### CODE REVIEW

File: src/test/java/org/kohsuke/github/GHContentIntegrationTest.java [hunk 1]
```diff
@@ -170,4 +171,25 @@ public void testGetFileContentWithNonAsciiPath() throws Exception {
...
+        // but symlinks to directories are symlinks!
+        assertThat(dirContent, allOf(hasProperty("target", is("a-dir-with-3-entries")),
+                                     hasProperty("type", is("symlink"))));
+        
+        // TODO somehow
+        /*
+        final GHContent fileContent2 = repo.getFileContent("ghcontent-ro/a-symlink-to-a-dir/entry-one");
+        // this needs special handling and will 404 from GitHub
+        assertThat(IOUtils.toString(fileContent.read(), StandardCharsets.UTF_8), is(""));
+        */
```
Comment (bitwiseman) [inline src/test/java/org/kohsuke/github/GHContentIntegrationTest.java:193]: There are ways to do this but they are inefficient/costly.

Comment (jtnord) [reply src/test/java/org/kohsuke/github/GHContentIntegrationTest.java:193]: agreed,  the API does not seem to support this, so given this is a mapping of the API to java I am not sure we should either.

That said, I was looking for this in order to solve https://issues.jenkins-ci.org/browse/JENKINS-62922 and if someone wants to acheive this I am not sure that making all users of the API handle it themselves is the correct approach.

**Comment by jtnord** [comment]
> Run mvn clean compile locally. This may reformat your code, commit those changes.

this changed every single file to windows line endings... I am pretty sure this is not what you want to see!

**Comment by jtnord** [comment]
this PR as is is enough to expose the target of a symlink.  

I have worked around the need to support traversal of symlinked directories and no longer have time to address this in this API (and am still unsure if this library should offer that or not).

Additionally GitHub support have not responded to my query about if there was a better API call (or set of calls) to achieve this than what has been discussed.

**Comment by jtnord** [comment]
build failures because CI is incorrectly setup.

```
Checking out the ref
  "C:\Program Files\Git\bin\git.exe" checkout --progress --force refs/remotes/pull/876/merge
  ##[error]error: unable to create file src/test/resources/org/kohsuke/github/GHContentIntegrationTest/wiremock/testGetFileContentWithSymlink/__files/repos_hub4j-test-org_ghcontentintegrationtest_contents_ghcontent-ro_a-symlink-to-a-dir-1baf6207-f25f-4064-af9b-8eb1cb1fe125.json: Filename too long
  ##[error]error: unable to create file 
```

`git config --system --add core.longpaths true`

**Comment by bitwiseman** [comment]
> > Run mvn clean compile locally. This may reformat your code, commit those changes.
> 
> this changed every single file to windows line endings... I am pretty sure this is not what you want to see!

Yeah, not as heavily tested on windows.

**Pull request merged**